breathecode/admissions/views.py

```diff
@@ -775,6 +775,14 @@ def delete(self, request, cohort_id=None, academy_id=None):
             items = Cohort.objects.filter(**lookups, academy__id=academy_id)
 
             for item in items:
+                item_users = CohortUser.objects.filter(
+                role=STUDENT,
```

Comment:
This code work, but is bad indented, and should be better check first all the items and them remove all, you can split it in two `for` loops, the serializers of DRF just do a change if all the input is rigth, think that some endpoint work checking the input before perform a change and other left pass some items, the user experience of user ( frontend ) that use we api it could be worse and confused

```py
for item in items:
    # do validations
    
for item in items:
    # perform deletion
```